README.md

```diff
@@ -1,15 +1,132 @@
+- `apache/*`
+- `github/*`
+- `actions/*`
+
+All other actions must be explicitly added to the allow list after undergoing a security review. This review process applies to both new actions and new versions of previously approved actions (though reviews for new versions are typically expedited).
+
+### Adding a New Action to the Allow List
+
+To request addition of an action to the allow list:
+
+1. **Fork** this repository
+2. **Add** an entry to `actions.yaml` using the following format:
+
+```yaml
+repo/owner:
+  '<exact-commit-sha>':
+    expires_at: 2050-01-01
+```
+
+3. **Create a PR** against the `main` branch
+4. **Include in your PR description**:
+   - Why this action is needed for your project
+   - Any alternatives you've considered
+   - Any security concerns you've identified
+
+5. **Wait for review** by the infrastructure team
+
+> [!NOTE]
+> Always pin actions to exact commit SHAs, never use tags or branch references.
+
+The infrastructure team will review your request and either approve, request changes, or provide feedback on alternatives.
+
+### Adding a new _version_ to the allow list
+
+In most cases, new versions are automatically added through Dependabot:
+- Dependabot opens PRs to update actions to the newest releases
+- The previously approved version will be marked to expire in 3 months
+- This grace period gives projects sufficient time to update their workflows
+
+### Manual Addition of Specific Versions
+
+If you need to add a specific version of an already approved action (especially an older one):
+
+1. **Fork** this repository
+2. **Add** a new version entry to an existing action in `actions.yaml` with the following format:
+
+```yaml
+existing/action:
+  '<exact-commit-sha>':
+    expires_at: 2025-01-01
+    keep: true
+```
+
+3. **Create a PR** against the `main` branch
+4. **Include in your PR description**:
+   - Specific reason why this version is required
+   - Any blockers preventing upgrade to newer versions
+   - Risk assessment for using an older version
+   - Expected timeline for migration to newer versions (if applicable)
+
+> [!WARNING]
+> Older versions may contain security vulnerabilities or performance issues. Always evaluate if using the latest version is possible before requesting older versions.
+
+### Removing a version manually
+
+> [!IMPORTANT]
+> If a version or entire action needs to be removed immediately due to a security vulnerability:
+
+1. **Fork** this repository
+2. **Remove** the relevant entry from `actions.yaml`
+3. **Create a PR** against the `main` branch
+4. **Mark it as urgent** in the PR title (e.g., "URGENT: Remove vulnerable action X")
+5. **Include in your PR description**:
+   - The reason for removal
+   - Any CVE or security advisory ID if applicable
+   - Impact on projects currently using the action
+   - Recommended alternatives if available
```

Comment:
The process was chatgpts idea but I do think it makes sense to have some process like this for cases like the `tj-actions` incident, where anyone who comes across a compromised actions can quickly initiate removal of said action.